File: src/lib/generateur_csv.cpp
#include "generateur_csv.hpp"

CsvGenerateur::CsvGenerateur()
	: CsvGenerateur(200)
{
	
}

CsvGenerateur::CsvGenerateur(unsigned int M_NOMBRE_ELEVE)
	: M_NOMBRE_ELEVE(M_NOMBRE_ELEVE),
	m_id_eleve(0)
{
	m_matiere_spe.SetSecurity(false);
	m_matiere_spe.Add("Francais");
	m_matiere_spe.Add("Anglais");
	m_prenom.SetSecurity(false);
	m_prenom.Add("Alice");
	m_prenom.Add("Antoine");
	m_nom.SetSecurity(false);
	m_nom.Add("Dupont");
	m_nom.Add("Ducon");

	M_NOMBRE_NOM = m_nom.GetNbElem();
	M_NOMBRE_PRENOM = m_prenom.GetNbElem();
	M_NOMBRE_MATIERE_SPE = m_matiere_spe.GetNbElem();
}
CsvGenerateur::~CsvGenerateur()
{
}

//std::string* PRENOM_TEMP[26] = { "Alice", "Bob", "Charlie", "David", "Eve", "Frank", "Grace", "Hannah", "Ivy", "Jack",
//"Katherine", "Liam", "Mia", "Noah", "Olivia", "Peter", "Quinn", "Rachel", "Samuel", "Taylor",
//"Ursula", "Victor", "Wendy", "Xavier", "Yasmine", "Zane" };


//std::array <std::string,29> const NOM = {"Dupont", "Martin", "Dubois", "Bernard", "Thomas", "Robert", "Richard", "Petit", "Durand", "Leroy",
//"Moreau", "Simon", "Laurent", "Lefebvre", "Michel", "Garcia", "David", "Roux", "Bonnet", "Fournier",
//"Lopez", "Legrand", "Martinez", "Ferreira", "Fernandez", "Blanc", "André", "Leroux", "Rousseau" };
//
//std::vector <std::string> MATIERE_SPE = { "Francais", "Histoire" "Geo", "EMC", "EPS", "SPC", "SVT", "Maths", "NSI", "Philo",
//"Anglais", "Espagnole", "Allemand", "Italien", "Russe" };


void CsvGenerateur::creerCsv()
{
	std::srand(static_cast<unsigned int>(std::time(nullptr)));
	std::ofstream myfile;
	myfile.open("eleve.csv");

	// Première ligne du csv avec les différentes catégories
	myfile << "ID;" << " PRENOM;" << "NOM;" << "Annee;";
	for (int i = 1; i < 11; i++)
	{
		myfile << "MATIERE OPTIONNELLE" << i << ";";
	};
	myfile << "\n";

	// Autres lignes du csv
	for (unsigned int i = 0; i < M_NOMBRE_ELEVE; i++)
	{
		myfile << m_id_eleve << ";";
		m_id_eleve++;
		myfile << m_prenom.Get(rand() % M_NOMBRE_PRENOM) << ";";
		myfile << m_nom.Get(rand() % M_NOMBRE_NOM) << ";";
		myfile << rand() % 3 + 1 << ";";

		NOMBRE_MATIERE_SPE = rand() % m_matiere_spe.GetNbElem();
		TabDynString m_matiere_restante(m_matiere_spe);
		for (unsigned int e = 0; e < NOMBRE_MATIERE_SPE +1; e++)

		{
			unsigned int RANDOM_MATIERE = rand() % m_matiere_restante.GetNbElem();
			myfile << m_matiere_restante.Pop(RANDOM_MATIERE) << ";";
		};
		myfile << "\n";
	}
	myfile.close();
}

File: src/lib/generateur_csv.py
""" CSV generator """
from random import randint

NOMBRE_ELEVE = 200

prenoms = ["Alice", "Bob", "Charlie", "David", "Eve", "Frank", "Grace", "Hannah", "Ivy", "Jack",
"Katherine", "Liam", "Mia", "Noah", "Olivia", "Peter", "Quinn", "Rachel", "Samuel", "Taylor",
"Ursula", "Victor", "Wendy", "Xavier", "Yasmine", "Zane"]
noms_de_famille = ["Dupont", "Martin", "Dubois", "Bernard", "Thomas", "Robert", "Richard", "Petit", "Durand", "Leroy",
                   "Moreau", "Simon", "Laurent", "Lefebvre", "Michel", "Garcia", "David", "Roux", "Bonnet", "Fournier",
                   "Lopez", "Legrand", "Martinez", "Ferreira", "Fernandez", "Blanc", "André", "Leroux", "Rousseau"]
matieres_communes_1 = ["Francais", "Histoire" "Geo", "EMC", "EPS", "SPC", "SVT", "Maths","NSI","Philo","Anglais","Espagnole","Allemand","Italien","Russe"]

with open('eleve.csv', 'w') as eleve:
    eleve.write("Prenom;Nom;annee;")
    for i in range(1,11):
        eleve.write(f"matiere{i};")
    eleve.write("\n")
    for i in range(NOMBRE_ELEVE):
        eleve.write(f"{prenoms[randint(0,len(prenoms)-1)]};{noms_de_famille[randint(0,len(noms_de_famille)-1)]};")
        annee = randint(1,3)
        eleve.write(f"{annee};")
        matiere_remain = matieres_communes_1.copy()
        nombre_matiere = randint(0,10)
        for i in range(nombre_matiere):
            matiere_i = randint(0, len(matiere_remain)-1)
            matiere_temp = matiere_remain[matiere_i]
            matiere_remain.pop(matiere_i)
            eleve.write(f"{matiere_temp};")
        for i in range(10 - nombre_matiere):
            eleve.write("None;")
        eleve.write("\n")

